storm-core/src/jvm/org/apache/storm/daemon/supervisor/workermanager/DefaultWorkerManager.java

```diff
@@ -0,0 +1,408 @@
+                        // ignore
+                    }
+                }
+            }
+            LOG.info("Shut down {}:{}", supervisorId, workerId);
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+        return null;
+    }
+
+    @Override
+    public boolean cleanupWorker(String workerId) {
+        try {
+            //clean up for resource isolation if enabled
+            if (resourceIsolationManager != null) {
+                resourceIsolationManager.releaseResourcesForWorker(workerId);
+            }
+            //Always make sure to clean up everything else before worker directory
+            //is removed since that is what is going to trigger the retry for cleanup
+            String workerRoot = ConfigUtils.workerRoot(conf, workerId);
+            if (Utils.checkFileExists(workerRoot)) {
+                if (runWorkerAsUser) {
+                    SupervisorUtils.rmrAsUser(conf, workerId, workerRoot);
+                } else {
+                    Utils.forceDelete(ConfigUtils.workerHeartbeatsRoot(conf, workerId));
+                    Utils.forceDelete(ConfigUtils.workerPidsRoot(conf, workerId));
+                    Utils.forceDelete(ConfigUtils.workerTmpRoot(conf, workerId));
+                    Utils.forceDelete(ConfigUtils.workerRoot(conf, workerId));
+                }
+                ConfigUtils.removeWorkerUserWSE(conf, workerId);
+            }
+            return true;
+        } catch (IOException e) {
+            LOG.warn("Failed to cleanup worker {}. Will retry later", workerId, e);
+        } catch (RuntimeException e) {
+            LOG.warn("Failed to cleanup worker {}. Will retry later", workerId, e);
+        }
+        return false;
+    }
+
+    @Override
+    public IWorkerResult resizeWorker(String supervisorId, String assignmentId, String stormId, Long port, String workerId, WorkerResources resources) {
+        return null;
+    }
+
+    protected String jlp(String stormRoot, Map conf) {
+        String resourceRoot = stormRoot + Utils.FILE_PATH_SEPARATOR + ConfigUtils.RESOURCES_SUBDIR;
+        String os = System.getProperty("os.name").replaceAll("\\s+", "_");
+        String arch = System.getProperty("os.arch");
+        String archResourceRoot = resourceRoot + Utils.FILE_PATH_SEPARATOR + os + "-" + arch;
+        String ret = archResourceRoot + Utils.CLASS_PATH_SEPARATOR + resourceRoot + Utils.CLASS_PATH_SEPARATOR + conf.get(Config.JAVA_LIBRARY_PATH);
+        return ret;
+    }
+
+    protected String getWorkerClassPath(String stormJar, Map stormConf) {
+        List<String> topoClasspath = new ArrayList<>();
+        Object object = stormConf.get(Config.TOPOLOGY_CLASSPATH);
+
+        if (object instanceof List) {
+            topoClasspath.addAll((List<String>) object);
+        } else if (object instanceof String) {
+            topoClasspath.add((String) object);
+        } else {
+            LOG.error("topology specific classpath is invaild");
```

Comment:
If object is null this will run, and it is not actually an error, so we probably should not output an error message.